crypto/x509/x_attrib.c

```diff
@@ -56,3 +57,228 @@ X509_ATTRIBUTE *X509_ATTRIBUTE_create(int nid, int atrtype, void *value)
     ASN1_TYPE_free(val);
     return NULL;
 }
+
+static int print_hex(BIO *out, unsigned char *buf, int len)
+{
+    int i;
+
+    for (i = 0; i < len; i++) {
+        if (BIO_printf(out, "%02X ", buf[i]) <= 0) {
+            return 0;
+        }
+    }
```

Comment:
This is minor, but since we have OPENSSL_buf2hexstr_ex already available, it may be worth leveraging that here.  Its minor though, may not be worth doing.